### ISSUE TITLE
EC2 controller throws `resource reference wrapper or ID required: GroupName,GroupRef` when referencing a SecurityGroup using `groupID` in `userIDGroupPairs`

### ISSUE THREAD

**Issue by wjf3121** [aws-controllers-k8s/community#2068]
**Describe the bug**
This bug is a regression observed since Ec2 controller v1.2.6, the following resource will throw `resource reference wrapper or ID required: GroupName,GroupRef` error:

```
apiVersion: ec2.services.k8s.aws/v1alpha1
kind: SecurityGroup
metadata:
  annotations:
    services.k8s.aws/region: us-east-1
  name: testsg
spec:
  description: test security group
  egressRules:
  - fromPort: 0
    ipProtocol: "-1"
    toPort: 0
    userIDGroupPairs:
    - description: allow all traffic
      groupID: sg-0xxxxxxxxxxxxxxxxxx
  ingressRules:
  - fromPort: 0
    ipProtocol: "-1"
    toPort: 0
    userIDGroupPairs:
    - description: allow all traffic
      groupID: sg-0xxxxxxxxxxxxxxxxxx
  name: testsg
  tags:
  - key: Project
    value: Test
  vpcID: vpc-xxxxxxxxxxxxxx
```

The resource can be successfully reconciled after downgrading EC2 controller to v1.2.5

**Steps to reproduce**
See bug description

**Expected outcome**
Security groups with rules referring to other source/dest security groups using groupID should succeed.

**Environment**

* Kubernetes version: 1.27
* Using EKS (yes/no): yes, 1.26
* AWS service targeted (S3, RDS, etc.): EC2 (security group)

**Comment by wjf3121** [comment]
Hi @a-hilaly , looks like it's caused by https://github.com/aws-controllers-k8s/ec2-controller/pull/187 ?

**Comment by a-hilaly** [comment]
@wjf3121 i'm a shipping a fix in a bit

**Comment by a-hilaly** [comment]
@wjf3121 Can you confirm that you are not using any kind of references in these resources?

**Comment by wjf3121** [comment]
> @wjf3121 Can you confirm that you are not using any kind of references in these resources?

Yes I can confirm. The example in the description is exactly the one we reproduced the issue, with is IDs/labels redacted.

**Comment by wjf3121** [comment]
Kindly ping. @a-hilaly Is there any update to this issue? Happy to provide more debug info if needed:)

**Comment by a-hilaly** [comment]
@wjf3121 We just release v1.2.8 - it should fix the references issue. Can you give it a shot?

**Comment by wjf3121** [comment]
Thanks! Verified that the issue is fixed in v1.2.9. Appreciate it!
